packages/assertion-monitor/index.ts

```diff
@@ -0,0 +1,200 @@
+import * as fs from 'fs'
+import * as path from 'path'
+import { PublicClient, createPublicClient, http } from 'viem'
+import yargs from 'yargs'
+import { ChildNetwork as ChainInfo } from '../utils'
+import { nodeCreatedEventAbi } from './abi'
+import { getChainFromId, getDefaultBlockRange } from './chains'
+import { reportAssertionMonitorErrorToSlack } from './reportAssertionMonitorAlertToSlack'
+
+const CHUNK_SIZE = 800n
+
+const options = yargs(process.argv.slice(2))
+  .options({
+    fromBlock: { type: 'number', default: 0 },
+    toBlock: { type: 'number', default: 0 },
+    configPath: { type: 'string', default: 'config.json' },
+    enableAlerting: { type: 'boolean', default: false },
+  })
+  .strict()
+  .parseSync()
+
+const configFileContent = fs.readFileSync(
+  path.join(process.cwd(), options.configPath),
+  'utf-8'
+)
+const config = JSON.parse(configFileContent)
+
+if (!Array.isArray(config.childChains) || config.childChains.length === 0) {
+  console.error('Error: Chains not found in the config file.')
+  process.exit(1)
+}
+
+type ChunkProcessFunction<T> = (
+  fromBlock: bigint,
+  toBlock: bigint,
+  client: PublicClient
+) => Promise<T>
+// Updated processChunkedRange function
+async function processChunkedRange<T>(
+  fromBlock: bigint,
+  toBlock: bigint,
+  chunkSize: bigint,
+  client: PublicClient,
+  processChunk: ChunkProcessFunction<T>
+): Promise<T[]> {
+  const results: T[] = []
+
+  if (fromBlock === toBlock) {
+    const result = await processChunk(fromBlock, toBlock, client)
+    results.push(result)
+    return results
+  }
+
+  let currentFromBlock = fromBlock
+
+  while (currentFromBlock <= toBlock) {
+    const currentToBlock =
+      currentFromBlock + chunkSize - 1n < toBlock
+        ? currentFromBlock + chunkSize - 1n
+        : toBlock
+
+    const result = await processChunk(currentFromBlock, currentToBlock, client)
+    results.push(result)
+
+    if (currentToBlock === toBlock) break
+
+    currentFromBlock = currentToBlock + 1n
+  }
+
+  return results
+}
+const getBlockRange = async (
+  client: PublicClient,
+  childChainInfo: ChainInfo
+) => {
+  const latestBlockNumber = await client.getBlockNumber()
+  const toBlock = BigInt(options.toBlock) || latestBlockNumber
+
+  if (options.fromBlock) {
+    return { fromBlock: BigInt(options.fromBlock), toBlock }
+  }
+
+  const blockTime = getDefaultBlockRange(
```

Comment:
nitpick: `blockTime` is a bit misleading as a variable name here, since the name suggests it's dealing with times, when in reality it holds a count (number) of blocks.